apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx

```diff
@@ -1,49 +1,166 @@
 import React, {useState} from 'react';
 import PropTypes from 'prop-types';
 import {BodyTwoText, Heading1} from '@cdo/apps/componentLibrary/typography';
-import LearningGoalItem from './LearningGoalItem';
 import Button from '@cdo/apps/templates/Button';
+import {navigateToHref} from '@cdo/apps/utils';
+import RubricEditor from './RubricEditor';
+import {snakeCase} from 'lodash';
+
+const FORM_ID = 'rubrics-container';
+const RUBRIC_PATH = '/rubrics';
 
 export default function RubricsContainer({
   unitName,
   lessonNumber,
   levels,
   rubric,
+  lessonId,
 }) {
-  // note that the use of currentId here is temporary until we are connected to the data
-  const [currentId, setCurrentId] = useState(1);
-  const [learningGoalList, setLearningGoalList] = useState([{id: currentId}]);
+  const [learningGoalList, setLearningGoalList] = useState(
+    !!rubric
+      ? rubric.learningGoals
+      : [
+          {
+            key: 'learningGoal-1',
+            id: 'learningGoal-1',
+            learningGoal: '',
+            aiEnabled: false,
+          },
+        ]
+  );
+
+  const generateLearningGoalKey = () => {
+    let learningGoalNumber = learningGoalList.length + 1;
+    while (
+      learningGoalList.some(
+        learningGoal =>
+          learningGoal.key === `learningGoal-${learningGoalNumber}`
+      )
+    ) {
+      learningGoalNumber++;
+    }
+
+    return `learningGoal-${learningGoalNumber}`;
+  };
+
+  const addNewConceptHandler = event => {
+    event.preventDefault();
+
+    const newKey = generateLearningGoalKey();
+    const newId = newKey;
+    const startingData = {
+      key: newKey,
+      id: newId,
+      learningGoal: '',
+      aiEnabled: false,
+    };
```

Comment:
would it be possible to extract this into a function and also use it to instantiate the first learning goal?

Reply:
I had some trouble with this because the function depends on knowing the length of this list before the list is created...  I am planning to chat about it tomorrow.